### DISCUSSION TITLE
Recovery after downscaling EKS cluster

### DISCUSSION THREAD

**Discussion by VijayDeyTR** [milvus-io/milvus#23131] [Q&A and General discussion]
Hi I have a question regarding how Milvus recovers on restarting. I have deployed a Milvus cluster on AWS EKS using S3 as the object storage using Helm.

I want to run hydrate data, run some experiments and then downscale the EKS nodegroups to 0 instances to save costs on ideal times. Would the previous data be recovered when we I bring up the nodes and deploy milvus again?

On a similar note, are there best practices for Disaster recovery for Milvus?

**Comment by LoveEachDay** [comment]
Yes, you can. You'd also enable storageclass to keep data for etcd and pulsar/kafka persisted, so that you can recovery a milvus while adding instances to EKS nodegroup.

**Reply by VijayDeyTR** [reply]
We have a enabled storage class for persistence.  We tried testing this scenario by following steps:
1. Created the stack using `helm upgrade --install` and hydrated data.
2. Deleted the stack using `helm uninstall milvus` (retained the PersistenceVolume claims and the corressponding EBS volumes) and downscaled the nodegroup in EKS to 0 instance.
3. Upscaled the nodegroup to desired instance and re-installed app using `helm upgrade --install`

However the pods etcd failed to restart with below error. Could you please help with this or share some resource on how to recover after we EKS nodes are brought down?
![Screen Shot 2023-03-29 at 1 55 12 PM](https://user-images.githubusercontent.com/90347493/228846701-72784a4b-d281-4554-840f-938175906fdc.png)

**Reply by LoveEachDay** [reply]
Could you provide the complete log output for a etcd pod? And also provide the output of `kubectl describe <etcd-pod>` .

**Reply by VijayDeyTR** [reply]
When trying to replicate the same scenario to get logs, this time the same error has occurred for ZK. Unfortunately ZK has no logs.
![Screen Shot 2023-03-31 at 5 50 18 PM](https://user-images.githubusercontent.com/90347493/229239495-e89aaf16-7c23-4ca8-8149-e1022aa88c90.png)
